### ISSUE TITLE
Agent vulnerability information showing in the vulnerability dashboard even after removing the agent.

### ISSUE THREAD

**Issue by sakib789** [wazuh/wazuh#25174]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 | vulnerability detector  | - | - | Ubuntu22.04 |

Hi Team,

While working with vulnerabilities, I noticed that Wazuh does not remove the agent and its related vulnerabilities from the Vulnerability Dashboard after completely removing the agent from the Wazuh Manager and restarting the Manager. I am sharing some screenshots for reference:

**Before removing the agent**
![image](https://github.com/user-attachments/assets/11d92e2c-3660-4717-bcc0-52a0ebe25225)

**After removing the agent**
![image](https://github.com/user-attachments/assets/e87d3f6f-1706-4128-be94-10667df70cff)

Please look into this issue.

**Comment by Dwordcito** [comment]
@sebasfalcone time box: 2 days.

@sakib789 could you expand the information on this issue? we need the reproduction steps or some environment with this problem to follow with a fix.

**Comment by sakib789** [comment]
Hi @sebasfalcone 

I tested in another environment in 4.8.1. The vulnerability Dasbhbaord is working properly in that deployment. But I am still facing this in 4.8.0. 

To reproduce the issue after stopping the agent server. I removed an agent from the Wazuh manager(v4.8.0) and restarted the Wazuh manager service. Chcek the screenshot for reference.

![image](https://github.com/user-attachments/assets/49c46763-4cc4-42e4-aae8-0ea57621817f)


I believe in the first test on 4.8.1, I checked it before updating the dashboard. After that, I did multiple tests in 4.8.1 and I have not faced the same issue. 
![image](https://github.com/user-attachments/assets/3dfb155e-ceb7-4f1e-8b39-27d46c675992)

 
As it is working properly in the latest version we can close this issue.
Thank you for your time.
